src/liballoc/collections/btree/search.rs

```diff
@@ -51,7 +51,7 @@ where
 /// exist in the node itself, it may exist in the subtree with that index
 /// (if the node has subtrees). If the key doesn't exist in node or subtree,
 /// the returned index is the position or subtree to insert at.
-pub fn search_linear<BorrowType, K, V, Type, Q: ?Sized>(
```

Comment:
Could we move/copy the doc comment here onto search_range since that's the common "entry point" now?

Doesn't have to be in this PR.

Reply:
Right, though I couldn't find any other documentation describing a returned enum (other than Result or Option).

Reply:
Uh, not sure how that's relevant? We rarely encounter returned enums in the wild, I think (though I'm sure now that I've said it I'll think of dozens of examples once I post this).

If you're looking for a format or so to follow, I wouldn't worry that much about that. I'd be far more interested in just some docs pointing out what the variants mean; basically same as the docs here do wrt to false/true and the index, but applied to the enum.

Reply:
I found it hard to describe without mentioning the variant (or at least, the GoDown variant which is somewhat confusing).

Reply:
Mentioning the variant is good (and indeed I would expect the docs to do so).